extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/ColorSplasher.pushbutton/script.py

```diff
@@ -110,52 +89,51 @@ def Execute(self, uiapp):
+                        if version > 2021:
+                            if str(wndw.crt_view.GetColorFillSchemeId(sel_cat._cat.Id)) == "-1":
+                                fColorScheme = DB.FilteredElementCollector(new_doc).OfClass(DB.BuiltInCategoryFillScheme).ToElements()
+                                if len(fColorScheme) > 0:
+                                    for sch in fColorScheme:
+                                        if sch.CategoryId == sel_cat._cat.Id:
+                                            if len(sch.GetEntries()) > 0:
+                                                wndw.crt_view.SetColorFillSchemeId(sel_cat._cat.Id, sch.Id)
+                                                break
+                        else:
+                            wndw._txtBlock5.Visible = True
                     else:
-                        wndw._txtBlock5.Visible = True
-                else:
-                    wndw._txtBlock5.Visible = False
-                    
-                for indx in range(wndw._listBox2.Items.Count):
-                    ogs = OverrideGraphicSettings().Dispose()
-                    ogs = OverrideGraphicSettings()
-                    color = Autodesk.Revit.DB.Color(wndw._listBox2.Items[indx]['Value']._n1, wndw._listBox2.Items[indx]['Value']._n2, wndw._listBox2.Items[indx]['Value']._n3)
-                    ogs.SetProjectionLineColor(color)
-                    ogs.SetSurfaceForegroundPatternColor(color);
-                    ogs.SetCutForegroundPatternColor(color);
-                    if solidFillId != None:
-                        ogs.SetSurfaceForegroundPatternId(solidFillId);
-                        ogs.SetCutForegroundPatternId(solidFillId)
-                    ogs.SetProjectionLinePatternId(ElementId(-1));
-                    for id in wndw._listBox2.Items[indx]['Value']._eleId:
-                        view.SetElementOverrides(id, ogs)
-                t.Commit()
+                        wndw._txtBlock5.Visible = False
+
+                    for indx in range(wndw._listBox2.Items.Count):
+                        ogs = DB.OverrideGraphicSettings().Dispose()
+                        ogs = DB.OverrideGraphicSettings()
+                        color = DB.Color(wndw._listBox2.Items[indx]['Value']._n1, wndw._listBox2.Items[indx]['Value']._n2, wndw._listBox2.Items[indx]['Value']._n3)
+                        ogs.SetProjectionLineColor(color)
+                        ogs.SetSurfaceForegroundPatternColor(color)
+                        ogs.SetCutForegroundPatternColor(color)
+                        if solidFillId is not None:
+                            ogs.SetSurfaceForegroundPatternId(solidFillId)
+                            ogs.SetCutForegroundPatternId(solidFillId)
+                        ogs.SetProjectionLinePatternId(DB.ElementId(-1))
+                        for id in wndw._listBox2.Items[indx]['Value']._eleId:
```

Comment:
Not tested with Forms controls Items, but can this just be

```
from item in wndw._listBox2.Items:
    value = item["Value"]
```

so we avoid the repetion of accessing the list by index every time?